src/ui/utils/testRuns.ts

```diff
@@ -77,34 +77,45 @@ export function groupRecordings(recordings: Recording[]): RecordingGroups {
   for (const filePath in recordingsMap) {
     const recordings = recordingsMap[filePath];
 
+    const didAnyTestFail = recordings.some(testFailed);
     const didAnyTestPass = recordings.some(testPassed);
 
+    function addToRecordingGroup(group: RecordingGroup, recording: Recording) {
+      if (group.fileNameToRecordings[filePath]) {
+        group.fileNameToRecordings[filePath].push(recording);
+      } else {
+        group.count++;
+        group.fileNameToRecordings[filePath] = [recording];
+      }
+    }
+
     for (const recording of recordings) {
       if (testPassed(recording)) {
-        if (passedRecordings.fileNameToRecordings[filePath]) {
-          passedRecordings.fileNameToRecordings[filePath].push(recording);
+        if (didAnyTestFail) {
+          addToRecordingGroup(flakyRecordings, recording);
         } else {
-          passedRecordings.count++;
-          passedRecordings.fileNameToRecordings[filePath] = [recording];
+          addToRecordingGroup(passedRecordings, recording);
         }
       } else if (testFailed(recording)) {
         if (didAnyTestPass) {
-          if (flakyRecordings.fileNameToRecordings[filePath]) {
-            flakyRecordings.fileNameToRecordings[filePath].push(recording);
-          } else {
-            flakyRecordings.count++;
-            flakyRecordings.fileNameToRecordings[filePath] = [recording];
-          }
+          addToRecordingGroup(flakyRecordings, recording);
         } else {
-          if (failedRecordings.fileNameToRecordings[filePath]) {
-            failedRecordings.fileNameToRecordings[filePath].push(recording);
-          } else {
-            failedRecordings.count++;
-            failedRecordings.fileNameToRecordings[filePath] = [recording];
-          }
+          addToRecordingGroup(failedRecordings, recording);
         }
       }
     }
+
+    if (didAnyTestFail && didAnyTestPass) {
+      flakyRecordings.fileNameToRecordings[filePath].sort((recordingA, recordingB) => {
```

Comment:
Why isn't sorting by recording date/time sufficient? (Do we not have that info loaded?)

Reply:
IIRC the recording date/time is the time when the upload started and (again, IIRC) that's not reliably in the same order.

Reply:
Ooof. That's fair.